SETUP/upgrade/13/20190929_update_user_project_info.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+$relPath='../../../pinc/';
+include_once($relPath.'base.inc');
+include($relPath.'udb_user.php');
+
+header('Content-type: text/plain');
+
+// ------------------------------------------------------------
+
+echo "Updating user_project_info to set t_latest_page_event=0 for projects \n";
+echo "where the user has only checked out or returnToRound a page...\n\n";
+
+// Load a list of all users
+$sql = "
+    SELECT username
+    FROM users
+    ORDER BY username
+";
+
+$result = mysqli_query(DPDatabase::get_connection(), $sql)
+    or die( mysqli_error(DPDatabase::get_connection()) );
+
+$users = [];
+while($row = mysqli_fetch_assoc($result))
+{
+    $users[] = $row['username'];
+}
+mysqli_free_result($result);
+
+// Loop though all users getting their user_project_info records to see what
+// projects they have "worked on" (ie: t_latest_page_event > 0)
+foreach($users as $username)
+{
+    echo "Processing $username...\n";
+
+    // load all projects this user has "worked on"
+    $sql = "
+        SELECT projectid
+        FROM user_project_info
+        WHERE
+            username='$username'
+            AND t_latest_page_event > 0
+    ";
+
+    $result = mysqli_query(DPDatabase::get_connection(), $sql)
+        or die( mysqli_error(DPDatabase::get_connection()) );
+
+    $projectids = [];
+    while($row = mysqli_fetch_assoc($result))
+    {
+        $projectids[] = $row['projectid'];
+    }
+    mysqli_free_result($result);
+
+    // for each project, get unique page_events for the project, if the only
```

Comment:
I suggest moving this comment to *inside* the `foreach` loop. Then make it something like:
```
Consider all of this user's page events for this project,
and fetch just the distinct event_type values for those events.
If the only event_types are (at most) 'checkedout' and 'returnToRound',
then this is a case where the new code in pinc/DPage.inc,
had it been in place back then, would never have updated t_latest_page_event.
So set user_project_info.t_latest_page_event=0 for this (username, projectid),
to achieve the same effect retroactively.
```

Reply:
I've update the wording based largely on what you have here. I am leaving it outside the foreach, however, as that's how documenting blocks makes the most sense to me (I know you and I differ on this).